pkg/webhook/admission/servingruntime/servingruntime_webhook.go

```diff
@@ -170,3 +199,60 @@ func validateServingRuntimePriority(newSpec *v1alpha1.ServingRuntimeSpec, existi
 	}
 	return nil
 }
+
+// validateMultiNodeSpec validates one of the following: tensor-parallel-size, pipeline-parallel-size, or WorkerSpec.PipelineParallelSize
+func validateMultiNodeSpec(newSpec *v1alpha1.ServingRuntimeSpec, existingSpec *v1alpha1.ServingRuntimeSpec) error {
+	// new sr,csr can not remove workerSpec in existing one
+	if existingSpec.WorkerSpec != nil && newSpec.WorkerSpec == nil {
+		return errors.New(DisallowedRemovingWorkerSpecFromServingRuntimeError)
+	}
+
+	if newSpec.WorkerSpec != nil {
+		if len(newSpec.WorkerSpec.Containers) > 1 {
+			return errors.New(DisallowedMultipleContainersInWorkerSpecError)
+		}
+
+		for i, container := range newSpec.Containers {
+			if container.Name == constants.InferenceServiceContainerName {
+				if _, exists := utils.GetEnvVarValue(newSpec.Containers[i].Env, constants.PipelineParallelSizeEnvName); exists {
+					return errors.New(DisallowedWorkerSpecPipelineParallelSizeEnvError)
+				}
+
+				if _, exists := utils.GetEnvVarValue(newSpec.Containers[i].Env, constants.TensorParallelSizeEnvName); exists {
+					return errors.New(DisallowedWorkerSpecTensorParallelSizeEnvError)
+				}
+				if utils.IsUnknownGpuResourceType(container.Resources) {
+					return fmt.Errorf(InvalidUnknownGPUTypeError, constants.InferenceServiceContainerName)
+				}
+			}
+		}
+		for _, container := range newSpec.WorkerSpec.Containers {
```

Comment:
isn't `newSpec.WorkerSpec.Containers` supposed to be 1?
On line 212 an error will be returned in case it is greater than 1.

Reply:
oh good catch.updated